DiffPlex.Wpf/Controls/InternalLinesViewer.xaml.cs

```diff
@@ -68,7 +70,151 @@ public void Clear()
+            grid.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Auto });
+            grid.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Auto });
+            grid.ColumnDefinitions.Add(new ColumnDefinition { Width = new GridLength(1, GridUnitType.Star) });
+
+            var index = new TextBlock
+            {
+                Text = number.HasValue ? number.ToString() : string.Empty,
+                TextAlignment = TextAlignment.Right,
+                VerticalAlignment = VerticalAlignment.Top,
+                Margin = new Thickness(2)
+            };
+            index.SetBinding(TextBlock.ForegroundProperty, GetBindings("LineNumberForeground", diffViewer, Foreground));
+            index.SetBinding(TextBlock.BackgroundProperty, GetBindings(changeType + "Background", diffViewer));
+            ApplyTextBlockProperties(index, diffViewer);
+
+            var op = new TextBlock
+            {
+                Text = operation,
+                TextAlignment = TextAlignment.Center,
+                VerticalAlignment = VerticalAlignment.Top,
+                Margin = new Thickness(2)
+            };
+            op.SetBinding(TextBlock.ForegroundProperty, GetBindings("ChangeTypeForeground", diffViewer, Foreground));
+            op.SetBinding(TextBlock.BackgroundProperty, GetBindings(changeType + "Background", diffViewer));
+            ApplyTextBlockProperties(op, diffViewer);
+
+            Grid.SetColumn(index, 0);
+            Grid.SetColumn(op, 1);
+            grid.Children.Add(index);
+            grid.Children.Add(op);
+
+            var text = new TextBlock
+            {
+                Text = value,
+                TextWrapping = IsTextWrapEnabled ? TextWrapping.Wrap : TextWrapping.NoWrap,
+                VerticalAlignment = VerticalAlignment.Top,
+                Margin = new Thickness(5, 0, 5, 0)
+            };
+            if (!string.IsNullOrEmpty(value))
+            {
+                text.SetBinding(TextBlock.ForegroundProperty, GetBindings(changeType + "Foreground", diffViewer, Foreground));
+                text.SetBinding(TextBlock.BackgroundProperty, GetBindings(changeType + "Background", diffViewer));
+                ApplyTextBlockProperties(text, diffViewer);
+                grid.ContextMenu = LineContextMenu;
+            }
+
+            Grid.SetColumn(text, 2);
+            grid.Children.Add(text);
+
+            panel.Children.Add(grid);
+
+            ValuePanel.Children.Add(panel);
+            ValuePanel.CanHorizontallyScroll = false;
+            ValueScrollViewer.HorizontalScrollBarVisibility = ScrollBarVisibility.Disabled;
+
+            return panel;
+        }
+        else
+        {
+            throw new Exception("Source is not a DiffViewer");
+        }
+    }
+
+    public WrapPanel Add(int? number, string operation, List<KeyValuePair<string, string>> value, string changeType, UIElement source)
+    {
+        if (source is DiffViewer diffViewer)
```

Comment:
Same comment here, we should support inlinedifferviewer and not do this check.

Reply:
This has been reverted. The clean up for some reason slowed the code down. I separated each operation into it's own method and simply checked if wrap or no wrap from a singular method. But there seems to be a loop that goes on for over 5 minutes creating all the textblocks